assets/src/apps/authoring/components/PropertyEditor/custom/CompJsonEditor.tsx

```diff
@@ -0,0 +1,77 @@
+import { selectCurrentSelection } from 'apps/authoring/store/parts/slice';
+import React, { CSSProperties, Fragment, useState } from 'react';
+import { useEffect } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import { useSelector } from 'react-redux';
+interface JsonEditorProps {
+  jsonValue: any;
+  onChange: (changedJson: any) => void;
+  schema: any;
+  existingPartIds: string[];
+}
+const CompJsonEditor: React.FC<JsonEditorProps> = (props) => {
+  const { jsonValue, onChange, existingPartIds, schema } = props;
+  const val = { id: jsonValue.id, custom: jsonValue.custom };
+  const [value, setValue] = useState<string>(JSON.stringify(val, null, 4));
+  const [validationMsg, setValidationMsg] = useState<string>('');
+  const currentPartSelection = useSelector(selectCurrentSelection);
+  const [displayEditor, setDisplayEditor] = useState<boolean>(false);
+  const textAreaStyle: CSSProperties = {
+    width: '100%',
+  };
+  useEffect(() => {
+    try {
+      const jsonVal = JSON.parse(value);
+      if (existingPartIds.indexOf(jsonVal.id) !== -1 && currentPartSelection !== jsonVal.id) {
+        setValidationMsg('ID you have used is already exist in the current Activity.');
+      } else {
+        setValidationMsg(jsonVal.id === '' ? 'ID is required Field and can not be empty' : '');
```

Comment:
it's a little more verbose but I think it'd be easier to follow if you split these
```
} else if (!jsonVali.id) {
  setValidationMsg('ID is required and cannot be empty');
} else {
  setValidationMsg('');
}
```